src/App.js

```diff
@@ -0,0 +1,65 @@
+import React, { Fragment, useState, useEffect } from 'react';
+import { isExpired, decodeToken } from 'react-jwt';
+
+import { login } from './methods/user/user';
+
+/*
+  These component could be lazily loaded, e.g.
+  const Login = React.lazy(() => import('./components/login/login'));
+  To allow for fast showing of the login form
+  (I couldn't quickly get Suspense to work with the testing library)
+*/
+import Application from './components/application/application';
+import Login from './components/login/login';
+
+function App() {
+  const [user, setUser] = useState(false);
+  const [loginError, setLoginError] = useState(false);
+  const [hasCheckedUser, setHasCheckedUser] = useState(false);
+  // Read token from local storage on first render
+  useEffect(() => {
+    const token = localStorage.getItem('user');
+    if (token) {
+      const decodedToken = decodeToken(token);
+      // TODO: Check token expiry/validity
+      if (decodedToken && decodedToken.email) {
+        setUser({
+          email: decodedToken.email
+        });
+      }
+    }
+    setHasCheckedUser(true);
+  }, []);
+
+  const handleLogin = (email, password) => {
+    setLoginError(false);
+    login(email, password).then(
```

Comment:
no love for `async` ?

Reply:
Good spot, I do like async. I have updated and it is cleaner :)